update-engine/src/context.rs

```diff
@@ -60,28 +71,45 @@ impl<S: StepSpec> StepContext<S> {
         report: EventReport<S2>,
     ) -> Result<(), NestedError> {
         let mut res = Ok(());
-        for event in report.step_events {
-            if let StepEventKind::ExecutionFailed { message, causes, .. } =
-                &event.kind
-            {
-                res = Err(NestedError::new(message.clone(), causes.clone()));
+        let delta_report = if let Some(id) = report.root_execution_id {
+            let mut nested_buffers = self.nested_buffers.lock().unwrap();
+            Some(nested_buffers.entry(id).or_default().add_event_report(report))
+        } else {
+            // If there's no root execution ID set, report is expected to be
+            // empty. However, report is untrusted data so we can't assert on
+            // it. Instead, log this.
```

Comment:
This is not a case that we need to handle, but again just checking my understanding: if we got reports from an out-of-date installinator we'd hit this case, right? It would send events but wouldn't know about the new `root_execution_id` field?

Reply:
Yes, that would be a problem, and I'd do this differently if this came up after we shipped.